### PR TITLE
Upgrade to mp rest client api101

### PR DESCRIPTION
**Pull request by andymc12**
Resolves issue #2258

### CODE REVIEW

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
Failures, but none are related to these changes (which would only affect the MP Rest Client projects).

File: dev/com.ibm.websphere.org.eclipse.microprofile.rest.client.1.0/bnd.bnd [hunk 1]
```diff
@@ -17,11 +17,11 @@ javac.target: 1.8
...
-  org.eclipse.microprofile.rest.client;version=1.0,\
-  org.eclipse.microprofile.rest.client.annotation;version=1.0,\
-  org.eclipse.microprofile.rest.client.ext;version=1.0,\
-  org.eclipse.microprofile.rest.client.inject;version=1.0,\
-  org.eclipse.microprofile.rest.client.spi;version=1.0
+  org.eclipse.microprofile.rest.client;version=1.0.1,\
+  org.eclipse.microprofile.rest.client.annotation;version=1.0.1,\
+  org.eclipse.microprofile.rest.client.ext;version=1.0.1,\
+  org.eclipse.microprofile.rest.client.inject;version=1.0.1,\
+  org.eclipse.microprofile.rest.client.spi;version=1.0.1
```
Comment (tevans78) [inline dev/com.ibm.websphere.org.eclipse.microprofile.rest.client.1.0/bnd.bnd]: It seemed odd to me that all of these packages would have changed (package version is not the same as api version). I went a looked in the eclipse repo and I'm struggling to find where the correct versions are. Only one I could find was for the top level package, and that still says `1.0`...
https://github.com/eclipse/microprofile-rest-client/blob/1.0.1/api/src/main/java/org/eclipse/microprofile/rest/client/package-info.java
So probably both the version specified in the api is wrong and these versions are wrong. They need to be reconciled and made consistent but not sure how. @Emily-Jiang ??

File: dev/com.ibm.websphere.org.eclipse.microprofile.rest.client.1.0/bnd.bnd [hunk 2]
```diff
@@ -17,11 +17,11 @@ javac.target: 1.8
 Bundle-SymbolicName: com.ibm.websphere.org.eclipse.microprofile.rest.client.1.0; singleton:=true
 
 Export-Package: \
-  org.eclipse.microprofile.rest.client;version=1.0,\
-  org.eclipse.microprofile.rest.client.annotation;version=1.0,\
-  org.eclipse.microprofile.rest.client.ext;version=1.0,\
-  org.eclipse.microprofile.rest.client.inject;version=1.0,\
-  org.eclipse.microprofile.rest.client.spi;version=1.0
+  org.eclipse.microprofile.rest.client;version=1.0.1,\
```
Comment (Emily-Jiang) [inline dev/com.ibm.websphere.org.eclipse.microprofile.rest.client.1.0/bnd.bnd]: Why did you update the package version? Did you update any classes in this package? To me, it sounds like you use the bundle version for package version, which is incorrect.

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
changed the package versions back to 1.0

**Pull request merged**